notes/day-1.md

```diff
@@ -34,35 +34,55 @@
+  - [Kaggle](https://www.kaggle.com/) A source of open datasets that can be used on ML projects
 
 ## Modern Web Apps Performance - Filip Rakowski
-- Basically off screen components can be downloaded dynamically ([lazy loading modal](../photos/lazy-loading-modal.jpg))
-- [bundlephobia](https://bundlephobia.com/) - show the packages performances and alternatives
-- [Webpack Bundle Analyzer](https://www.npmjs.com/package/webpack-bundle-analyzer) to create visual dependency map
-- [import cost](https://github.com/wix/import-cost) to see the file size that will be added when you import a package
+- Rules for performance
+  - Split your code per route
+  - **Load off screen components lazily** ([lazy loading modal](../photos/lazy-loading-modal.jpg)), so the first load is as fast as possible
+  - Load non-critical libraries lazily
+  - Avoid bundling all 3rd party libs into one file. It's an anti pattern
+- Resources mentioned on the talk:
+  - [bundlephobia](https://bundlephobia.com/) - show the packages performances and alternatives
+  - [Webpack Bundle Analyzer](https://www.npmjs.com/package/webpack-bundle-analyzer) to create visual dependency map
+  - [import cost](https://github.com/wix/import-cost) to see the file size that will be added when you import a package
+  - [Code coverage](https://developers.google.com/web/updates/2017/04/devtools-release-notes#coverage) on Chrome to find where your bundles can be split
 
 ## GraphQL + Apollo + Vue - Sara Vieira
 - Apollo-queries wrap the components and the template needs to be wrapped with the result data with a slot-scope
+- Slides available here: https://vue-apollo-magic.now.sh/
+
+## Desktop applications with Vue
+- An overview of packages that can be used for Desktop apps using Vue
+- Can use not only [Electron](https://electronjs.org/), but [Nw.js](https://nwjs.io/) and [Vuido](https://github.com/mimecorg/vuido)
 
 ## Designing components in Vue - Damian Dulisz
-- prop-based solution
-  - low flexibility
-  - a lot of conditions in the templates
-  - hard to maintain
-- slot-based solution is the best solution in most cases
+- One approach to design components is **prop-based** solution
+  - Has low flexibility
+  - End up a lot of conditionals in the templates
+  - Hard to maintain
+- Slot-based solution is the best solution in most cases
+
+## Full Stack Vue in the Era of Serverless Computing - Nader Dabit
+- A glimpse on tools that AWS provides for serverless applications ([AWS Amplify](https://aws-amplify.github.io/))
+- Resource: https://medium.com/@dabit3/full-stack-development-in-the-era-of-serverless-computing-c1e49bba8580
 
 ## Design Patterns for Vue - Jacob Schatz
+- Vue Builder Pattern example for FormBuilder [see](https://codesandbox.io/s/rmmprvonnm)
 - [vuecommander.com](https://vuecommander.com/) for the command pattern
-- Vue builder pattern example [see](https://codesandbox.io/s/rmmprvonnm)
-
+- Recommended book: Design Patterns - Elements of Reusable Object-Oriented Software
```

Comment:
This is the book of Gang of Four. But he actually recommended: https://www.goodreads.com/book/show/43125355-dive-into-design-patterns

Reply:
Oh yes, indeed!